test_agent/rust/src/main.rs

```diff
@@ -44,19 +47,43 @@ fn connect_to_socket(addr: &str, port: u16) -> Result<TcpStream, Box<dyn std::er
     }
 }
 
-async fn connect_and_receive() -> Result<(), Box<dyn std::error::Error>> {
+async fn connect_and_receive(transport_name: &String) -> Result<(), Box<dyn std::error::Error>> {
     let test_agent = connect_to_socket(TEST_MANAGER_ADDR.0, TEST_MANAGER_ADDR.1)?;
     let ta_to_tm_socket = connect_to_socket(TEST_MANAGER_ADDR.0, TEST_MANAGER_ADDR.1)?;
     let foo_listener_socket_to_tm = connect_to_socket(TEST_MANAGER_ADDR.0, TEST_MANAGER_ADDR.1)?;
 
-    let u_transport = UTransportSocket::new()?;
-    debug!("Socket transport created successfully");
+    let u_transport: Box<dyn UTransport> = if "zenoh" == transport_name {
```

Comment:
I think using a match expression could help with the readability a bit here, e.g.
```rust
let u_transport: Box<dyn UTransport> = match transport_name {
  "zenoh" => create_zenoh_u_transport() // factor the code out for creating the transport
  _ => {
    dbg!("Socket transport created successfully");
    Box::new(UTransportSocket::new()?)
  }
}
```

Reply:
when trying this, clippy seems to prefer an "if" statement, due to it being an equality. Check it out and see if that's okay once I push